### PR TITLE
[JENKINS-29501] added support for crittercism plugin

### PR DESCRIPTION
**Pull request by marc-guenther**
added support for the Crittercism dSYM Plugin

### CODE REVIEW

**Comment by daspilker** [comment]
I have been rejecting pull requests which add DSL methods for setting plain text passwords. In this case the API key is plain text. I started a discussion on the mailing list about this, see https://groups.google.com/d/msg/job-dsl-plugin/59xhodIgrsY/L2_GrZNEBQAJ.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy [hunk 1]
```diff
@@ -1403,6 +1403,21 @@ class PublisherContext extends AbstractExtensibleContext {
     }
 
     /**
+     * @since 1.36
+     */
+    @RequiresPlugin(id = 'crittercism-dsym')
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy]: Add a minimum version.

File: docs/Job-reference.md [hunk 2]
```diff
@@ -5544,6 +5545,38 @@ job('example') {
...
+            apiKey(String apiKey = '')
+            appID(String appID = '')
+            filePath(String filePath = '')
+        }
+    }
+}
+```
+
+Adds a Crittercism dSYM Upload publisher. Requires the [Crittercism dSYM Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Crittercism+dSYM+Plugin).
+
```
Comment (daspilker) [inline docs/Job-reference.md]: Add a link to the new "Handling Credentials" page to give the user a hint about handling plain text credentials. See https://github.com/jenkinsci/job-dsl-plugin/blob/e917f45419ca2cbcf4cb95e673acdda6667eeeef/docs/Job-reference.md#hipchat-publisher for an example.

**Comment by daspilker** [comment]
Please rebase you changes to the current master branch.

**Comment by marc-guenther** [comment]
Just rebased, added the minimal plugin version (there has been only one release of the plugin so far, though), and added the link to the "Handling Credentials" page.

**Comment by daspilker** [comment]
Thanks! I had to rebased you changes again and merged them manually.

**Pull request closed without merging**